extensions/metrics-cluster-feature/renderer.tsx

```diff
@@ -1,55 +1,27 @@
-import { LensRendererExtension, Interface, Component, Catalog} from "@k8slens/extensions";
-import { MetricsFeature } from "./src/metrics-feature";
+import React from "react";
+import { LensRendererExtension, Catalog } from "@k8slens/extensions";
+import { MetricsSettings } from "./src/metrics-settings";
 
 export default class ClusterMetricsFeatureExtension extends LensRendererExtension {
-  onActivate() {
-    const category = Catalog.catalogCategories.getForGroupKind<Catalog.KubernetesClusterCategory>("entity.k8slens.dev", "KubernetesCluster");
-
-    if (!category) {
-      return;
-    }
-
-    category.on("contextMenuOpen", this.clusterContextMenuOpen.bind(this));
-  }
-
-  async clusterContextMenuOpen(cluster: Catalog.KubernetesCluster, ctx: Interface.CatalogEntityContextMenuContext) {
-    if (!cluster.status.active) {
-      return;
-    }
-
-    const metricsFeature = new MetricsFeature();
-
-    await metricsFeature.updateStatus(cluster);
-
-    if (metricsFeature.status.installed) {
-      if (metricsFeature.status.canUpgrade) {
-        ctx.menuItems.unshift({
-          icon: "refresh",
-          title: "Upgrade Lens Metrics stack",
-          onClick: async () => {
-            metricsFeature.upgrade(cluster);
-          }
-        });
-      }
-      ctx.menuItems.unshift({
-        icon: "toggle_off",
-        title: "Uninstall Lens Metrics stack",
-        onClick: async () => {
-          await metricsFeature.uninstall(cluster);
-
-          Component.Notifications.info(`Lens Metrics has been removed from ${cluster.metadata.name}`, { timeout: 10_000 });
+  entitySettings = [
+    {
+      apiVersions: ["entity.k8slens.dev/v1alpha1"],
+      kind: "KubernetesCluster",
+      title: "Lens Metrics",
+      priority: 5,
+      components: {
+        View: (props: {entity: Catalog.KubernetesCluster}) => { // eslint-disable-line
+          const cluster = props.entity; // eslint-disable-line
```

Comment:
If these disables are from what I think they are, I think we should just turn off that lint rule.

Reply:
It feels bad, yes. Not sure what's the proper way to fix this.

Reply:
The way I did it was to make it a free function:

```ts
interface EntitySettingViewProps {
  entity: Catalog.KubernetesCluster;
}

function Metrics({ entity }: EntitySettingViewProps) {
  return (
    <section>
      <section>
        <MetricsSettings cluster={entity} />
      </section>
    </section>
  );
}

export default class ClusterMetricsFeatureExtension extends LensRendererExtension {
  entitySettings = [
    {
      apiVersions: ["entity.k8slens.dev/v1alpha1"],
      kind: "KubernetesCluster",
      title: "Lens Metrics",
      priority: 5,
      components: {
        View: Metrics,
      },
    },
  ];
}
```

Reply:
Should be now ok.